bin/elixir

```diff
@@ -206,8 +206,8 @@ SCRIPT_PATH=$(dirname "$SELF")
 if [ "$OSTYPE" = "cygwin" ]; then SCRIPT_PATH=$(cygpath -m "$SCRIPT_PATH"); fi
 if [ "$MODE" != "iex" ]; then ERL="-noshell -s elixir start_cli $ERL"; fi
 
-if [ "$OS" != "Windows_NT" ]; then
-  if test -t 1 -a -t 2; then ERL="-elixir ansi_enabled true $ERL"; fi
+if [ "$OS" != "Windows_NT" -a -t 1 -a -t 2 -a "$NO_COLOR" != ""]; then
```

Comment:
I believe `-a -t` is not supported inside the `if` in all shell implementations. So please keep this part as it was. I also think the NO_COLOR spec says the variable should be respected regardless of value, so we should probably use `! -z` or similar.

Reply:
yeah, there's something wrong here because I'm getting this on macOS:

```
/Users/wojtek/src/elixir/bin/elixir: line 209: [: too many arguments
```

Reply:
These are as `test foo -a bar` and `[ foo -a bar ]` [are required by POSIX to be the same](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html#top)

Reply:
My understanding is with this patch we disable ANSI escape codes completely. However, ANSI besides colours have things like italics, underline etc - are these supposed to be disabled by NO_COLOR too?

Reply:
> These are as test foo -a bar and [ foo -a bar ] are required by POSIX to be the same

The problem is that `-t` is not in the standard and it fails in terms of syntax in some bashes but it is just fails in case of `test`.

Reply:
I have just checked, and it is in POSIX standard, so it should be supported in all `sh` that claim to be POSIX compliant.

@wojtekmach I missed the space before `]`.

Reply:
Running shellcheck on this file after changes emit these new warnings:

```
In bin/elixir line 4:
if [ $# -eq 0 -o "$1" = "--help" -o "$1" = "-h" ]; then
              ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.
                                 ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.

In bin/elixir line 209:
if [ "$OS" != "Windows_NT" -a -t 1 -a -t 2 -a ! -z "$NO_COLOR" ]; then
                           ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
                                   ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
                                           ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
                                              ^-- SC2236: Use -n instead of ! -z.


For more information:
  https://www.shellcheck.net/wiki/SC2166 -- Prefer [ p ] && [ q ] as [ p -a q...
```

Please address them.

Reply:
Fixed.